src/test/sighash_tests.cpp

```diff
@@ -244,6 +244,256 @@ BOOST_AUTO_TEST_CASE(sighash_from_data)
+        vsh.clear();
+        vScriptCode.clear();
+        // create a random tx and get the signature hashes
+        CMutableTransaction _tx;
+        RandomTransaction(_tx, false);
+        const CTransaction tx(_tx);
+        for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+            vScriptCode.emplace_back();
+            RandomScript(vScriptCode.back());
+            vsh.emplace_back(SignatureHash(vScriptCode.back(), tx, nIn, SIGHASH_ALL, 0, tx.GetRequiredSigVersion()));
+        }
+
+        // -- create malleated txes
+        CMutableTransaction mtx = _tx;
+        for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+            BOOST_CHECK(vsh[nIn] == SignatureHash(vScriptCode[nIn], CTransaction(mtx), nIn, SIGHASH_ALL, 0, tx.GetRequiredSigVersion()));
+        }
+
+        // remove one transparent input
+        if (mtx.vin.size() > 0) {
+            mtx.vin.pop_back();
+            CTransaction tx2(mtx);
+            for (int nIn = 0; nIn < (int) tx.vin.size() - 1; nIn++) {
+                BOOST_CHECK(vsh[nIn] != SignatureHash(vScriptCode[nIn], tx2, nIn, SIGHASH_ALL, 0, tx2.GetRequiredSigVersion()));
+            }
+        }
+
+        // change the prevout of the transparent input being signed
+        mtx = _tx;
+        for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+            while (mtx.vin[nIn].prevout.hash == tx.vin[nIn].prevout.hash)
+                mtx.vin[nIn].prevout.hash = InsecureRand256();
+            CTransaction tx2(mtx);
+            BOOST_CHECK(vsh[nIn] != SignatureHash(vScriptCode[nIn], tx2, nIn, SIGHASH_ALL, 0, tx2.GetRequiredSigVersion()));
+        }
+
+        // add one random transparent input
+        {
+            mtx = _tx;
+            CTxIn in(InsecureRand256(), InsecureRandBits(2));
+            RandomScript(in.scriptSig);
+            in.nSequence = (InsecureRandBool()) ? InsecureRand32() : (unsigned int)-1;
+            mtx.vin.emplace_back(in);
+            CTransaction tx2(mtx);
+            for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+                BOOST_CHECK(vsh[nIn] != SignatureHash(vScriptCode[nIn], tx2, nIn, SIGHASH_ALL, 0, tx2.GetRequiredSigVersion()));
+            }
+        }
+
+        // Shield inputs
+        if (tx.sapData && tx.sapData->vShieldedSpend.size() > 0) {
+            // remove one spend
+            mtx = _tx;
+            mtx.sapData->vShieldedSpend.pop_back();
+            for (int nIn = 0; nIn < (int) tx.vin.size(); nIn++) {
+                BOOST_CHECK(vsh[nIn] != SignatureHash(vScriptCode[nIn], CTransaction(mtx), nIn, SIGHASH_ALL, 0, SigVersion::SIGVERSION_SAPLING));
+            }
+
+            // modify one spend (cv, anchor, nullifier, rk, zkproof, spendAuthSig)
+            int i = InsecureRandRange(tx.sapData->vShieldedSpend.size());
+
+            mtx = _tx;
+            while (mtx.sapData->vShieldedSpend[i].cv == tx.sapData->vShieldedSpend[i].cv)
+                mtx.sapData->vShieldedSpend[i].cv = GetRandHash();
```

Comment:
I'm interested to know why did you added this, it's an overkill for something that should never happen.

Reply:
Same as all the other checks.